### PR TITLE
Fix audit report of BEP-127 and BEP-131

### PR DESCRIPTION
**Pull request by gothery001**

### CODE REVIEW

File: contracts/BSCValidatorSet.sol [hunk 1]
```diff
@@ -107,10 +107,11 @@ contract BSCValidatorSet is IBSCValidatorSet, System, IParamSubscriber, IApplica
   }
 
   modifier initValidatorExtraSet() {
-    if (validatorExtraSet.length == 0) {
+    uint256 len = currentValidatorSet.length;
```
Comment (realuncle) [inline contracts/BSCValidatorSet.sol]: Please do not use `len`, in some other language it is a native function.

File: contracts/BSCValidatorSet.sol [hunk 2]
```diff
@@ -491,7 +493,7 @@ contract BSCValidatorSet is IBSCValidatorSet, System, IParamSubscriber, IApplica
       || (maxNumOfMaintaining == 0 || maintainSlashScale == 0)      // - 1. check if not start
       || numOfMaintaining >= maxNumOfMaintaining                    // - 2. check if reached upper limit
       || !isWorkingValidator(index)                                 // - 3. check if not working(not jailed and not maintaining)
-      || validatorExtraSet[index].enterMaintenanceHeight > 0        // - 5. check if has Maintained
+      || validatorExtraSet[index].enterMaintenanceHeight > 0        // - 5. check if has Maintained during one 24-hour period
```
Comment (realuncle) [inline contracts/BSCValidatorSet.sol]: I am confused by `during one 24-hour period`

File: contracts/BSCValidatorSet.sol (hunk 1 above)
Comment (SolidityGo) [reply contracts/BSCValidatorSet.sol]: fixed

File: contracts/BSCValidatorSet.sol (hunk 2 above)
Comment (SolidityGo) [reply contracts/BSCValidatorSet.sol]: `EnterMaintenanceHeight` is cleared to zero in each 24-hour period,  `validatorExtraSet[index].enterMaintenanceHeight > 0` could be only used to check if has Maintained in current 24-hour period,  this is a more accurate description

File: contracts/BSCValidatorSet.sol [hunk 3]
```diff
@@ -732,7 +737,7 @@ contract BSCValidatorSet is IBSCValidatorSet, System, IParamSubscriber, IApplica
       }
 
       // record the jailed validator in validatorSet
-      for (uint k = 0; k < _validatorSet.length; k++) {
+      for (uint k = 0; k < _validatorSet.length; ++k) {
```
Comment (dean65) [inline contracts/BSCValidatorSet.sol]: should we keep the same style? `for (uint k; k < _validatorSet.length; ++k) {`

File: contracts/BSCValidatorSet.sol [hunk 4]
```diff
@@ -744,7 +749,7 @@ contract BSCValidatorSet is IBSCValidatorSet, System, IParamSubscriber, IApplica
     // 2. get unjailed validators from validatorSet
     unjailedValidatorSet = new Validator[](_validatorSet.length - numOfFelony);
     i = 0;
-    for (uint index = 0; index < _validatorSet.length; index++) {
+    for (uint index = 0; index < _validatorSet.length; ++index) {
```
Comment (dean65) [inline contracts/BSCValidatorSet.sol]: `for (uint index; index < _validatorSet.length; ++index) {` ?

File: contracts/BSCValidatorSet.sol [hunk 5]
```diff
@@ -813,7 +818,7 @@ contract BSCValidatorSet is IBSCValidatorSet, System, IParamSubscriber, IApplica
       } else if (idx == 1) {
         RLPDecode.RLPItem[] memory items = iter.next().toList();
         validatorSetPkg.validatorSet =new Validator[](items.length);
-        for (uint j = 0;j<items.length;j++) {
+        for (uint j = 0;j<items.length;++j) {
```
Comment (dean65) [inline contracts/BSCValidatorSet.sol]: `for (uint j;j<items.length;++j) {`?

File: contracts/BSCValidatorSet.sol (hunk 4 above)
Comment (gothery001) [reply contracts/BSCValidatorSet.sol]: fixed

File: contracts/BSCValidatorSet.sol (hunk 5 above)
Comment (gothery001) [reply contracts/BSCValidatorSet.sol]: fixed

File: contracts/BSCValidatorSet.sol (hunk 3 above)
Comment (gothery001) [reply contracts/BSCValidatorSet.sol]: fixed

**Review by realuncle — APPROVED** [review]

**Review by dean65 — APPROVED** [review]

**Review by axljoker03 — APPROVED** [review]

**Pull request merged**